### PR TITLE
added lesson 3: functions

### PR DESCRIPTION
**Pull request by 3tom**
@tomekDiakowTieto , please take a look.

Thanks!
Tomasz

### CODE REVIEW

File: students/medrek_tomasz/lesson_03_functions/exponentation.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
+#!/usr/bin/env python3
+
+
+def exponentiation(a, n):
```
Comment (tomekDiakowTieto) [inline students/medrek_tomasz/lesson_03_functions/exponentation.py]: FYI:
return 1 if n == 0 else a * exponentiation(a, n - 1)

But propably my solution will be unreadable, so this is only TIP for You

File: students/medrek_tomasz/lesson_03_functions/fibonacci_numbers.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+
+def fibonacci(n):
+    if (n == 0) or (n == 1):
+        return n
+    else:
+        return fibonacci(n - 1) + fibonacci(n - 2)
+
+
+try:
```
Comment (tomekDiakowTieto) [inline students/medrek_tomasz/lesson_03_functions/fibonacci_numbers.py]: You can put it inside while loop

File: students/medrek_tomasz/lesson_03_functions/exponentation.py (hunk 1 above)
Comment (3tom) [reply students/medrek_tomasz/lesson_03_functions/exponentation.py]: Nice tip, didn't know such thing is allowed, thanks.

File: students/medrek_tomasz/lesson_03_functions/fibonacci_numbers.py (hunk 2 above)
Comment (3tom) [reply students/medrek_tomasz/lesson_03_functions/fibonacci_numbers.py]: https://snakify.org/lessons/functions/problems/fibonacci_rec/
"Don't use loops, use the flair of recursion instead. However, you should think about why the recursive method is much slower than using loops. "

**Comment by 3tom** [comment]
@tomekDiakowTieto , sorry I missed it somehow. Already uploaded this calculator, please check.
Thanks!

File: students/medrek_tomasz/lesson_03_functions/calculator.py [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+        break
+
+    try:
+        x = float(input('Please enter first operand\n'))
+        y = float(input('Please enter second operand\n'))
+    except ValueError:
+        print('Not a valid, number, please try again\n')
+        continue
+
+    print('Result is:\n{}\n'.format(calculate(x, y, option)))
```
Comment (TomaszDiakowCoderslab) [inline students/medrek_tomasz/lesson_03_functions/calculator.py]: Looks great

Comment (3tom) [reply students/medrek_tomasz/lesson_03_functions/calculator.py]: It's old version o0
let me reupload

**Pull request merged**